arbos/l1pricing/l1pricing.go

```diff
@@ -191,19 +210,185 @@ func (ps *L1PricingState) SetPricePerUnit(price *big.Int) error {
 	return ps.pricePerUnit.Set(price)
 }
 
-func (ps *L1PricingState) L1BaseFeeEstimate() (*big.Int, error) {
-	perUnit, err := ps.pricePerUnit.Get()
+// Update the pricing model based on a payment by a batch poster
+func (ps *L1PricingState) UpdateForBatchPosterSpending(
+	statedb vm.StateDB,
+	evm *vm.EVM,
+	arbosVersion uint64,
+	updateTime, currentTime uint64,
+	batchPoster common.Address,
+	weiSpent *big.Int,
+	scenario util.TracingScenario,
+) error {
+	if arbosVersion < 2 {
+		return ps._preVersion2_UpdateForBatchPosterSpending(statedb, evm, updateTime, currentTime, batchPoster, weiSpent, scenario)
+	}
+
+	batchPosterTable := ps.BatchPosterTable()
+	posterState, err := batchPosterTable.OpenPoster(batchPoster, true)
+	if err != nil {
+		return err
+	}
+
+	fundsDueForRewards, err := ps.FundsDueForRewards()
+	if err != nil {
+		return err
+	}
+
+	// compute allocation fraction -- will allocate updateTimeDelta/timeDelta fraction of units and funds to this update
+	lastUpdateTime, err := ps.LastUpdateTime()
+	if err != nil {
+		return err
+	}
+	if lastUpdateTime == 0 && currentTime > 0 { // it's the first update, so there isn't a last update time
```

Comment:
I think this should be `updateTime > 0`. I've opened a PR to fix this and separate out the csv files into their own directory
- https://github.com/OffchainLabs/nitro/pull/761

Reply:
Thanks.  Merged that branch.